signature/policy_config.go

```diff
@@ -51,28 +52,44 @@ func (err InvalidPolicyFormatError) Error() string {
 // NOTE: When this function returns an error, report it to the user and abort.
 // DO NOT hard-code fallback policies in your application.
 func DefaultPolicy(sys *types.SystemContext) (*Policy, error) {
-	return NewPolicyFromFile(defaultPolicyPath(sys))
+	policyPath, err := defaultPolicyPath(sys)
+	if err != nil {
+		return nil, err
+	}
+	return NewPolicyFromFile(policyPath)
 }
 
 // defaultPolicyPath returns a path to the default policy of the system.
-func defaultPolicyPath(sys *types.SystemContext) string {
-	return defaultPolicyPathWithHomeDir(sys, homedir.Get())
+func defaultPolicyPath(sys *types.SystemContext) (string, error) {
+	policyFilePath, err := defaultPolicyPathWithHomeDir(sys, homedir.Get())
+	if err != nil {
+		return "", err
+	}
+	return policyFilePath, nil
 }
 
 // defaultPolicyPathWithHomeDir is an internal implementation detail of defaultPolicyPath,
 // it exists only to allow testing it with an artificial home directory.
-func defaultPolicyPathWithHomeDir(sys *types.SystemContext, homeDir string) string {
+func defaultPolicyPathWithHomeDir(sys *types.SystemContext, homeDir string) (string, error) {
+	var checkedPolicyPaths []string // To store the paths being checked
+
 	if sys != nil && sys.SignaturePolicyPath != "" {
-		return sys.SignaturePolicyPath
+		return sys.SignaturePolicyPath, nil
 	}
+
 	userPolicyFilePath := filepath.Join(homeDir, userPolicyFile)
+	checkedPolicyPaths = append(checkedPolicyPaths, userPolicyFilePath)
 	if err := fileutils.Exists(userPolicyFilePath); err == nil {
-		return userPolicyFilePath
+		return userPolicyFilePath, nil
 	}
 	if sys != nil && sys.RootForImplicitAbsolutePaths != "" {
-		return filepath.Join(sys.RootForImplicitAbsolutePaths, systemDefaultPolicyPath)
+		return filepath.Join(sys.RootForImplicitAbsolutePaths, systemDefaultPolicyPath), nil
+	}
+	checkedPolicyPaths = append(checkedPolicyPaths, systemDefaultPolicyPath)
+	if err := fileutils.Exists(systemDefaultPolicyPath); err == nil {
+		return systemDefaultPolicyPath, nil
 	}
-	return systemDefaultPolicyPath
+	return "", fmt.Errorf("no policy.json file found at any of the following:%s", strings.Join(checkedPolicyPaths, ", "))
```

Comment:
- Space after `:`
- It would be better to use `%q` for the individual paths, to avoid surprises.

Reply:
I fixed this please take a look now.